packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -742,7 +751,8 @@ class SemanticsData with Diagnosticable {
         && other.transform == transform
         && other.elevation == elevation
         && other.thickness == thickness
-        && _sortedListsEqual(other.customSemanticsActionIds, customSemanticsActionIds);
+        && _sortedListsEqual(other.customSemanticsActionIds, customSemanticsActionIds)
+        && other.headingLevel == headingLevel;
```

Comment:
maybe move this above the customaction for code style in this file0

Reply:
Done.